src/data-extractor/javaDataExtractor.ts

```diff
@@ -47,8 +47,10 @@ export class JavaDataExtractor extends AbstractDataExtractor {
     return { type: variable.type, name: variable.name, value: variable.value };
   }
 
-  createVariableId(variable: DebugProtocol.Variable): string {
-    return this.isNullVariable(variable) ? addNullPrefix(hash(variable)) : addObjectPrefix(variable.value.split(this.sizeSuffix)[0]);
+  createVariableId(variable: DebugProtocol.Variable, parentId?: string | undefined): string {
+    return this.isNullVariable(variable)
+      ? addNullPrefix(hash(parentId ? parentId : variable))
```

Comment:
The `parentId` is never undefined in this case. You can't have a null variable without a parent. Hashing of a string is also not necessary. Can be:
```
addNullPrefix(parentId)
```
or as mentioned [here](https://github.com/GinoCardillo-OST/Visual-OO-Debugger/pull/50/files#r895165462):
```
addNullPrefix(`${parentId}_${variableName}`)
```

Reply:
Hashing is indeed not necessary anymore but the check if the parent id is null needs to be done because of the linting rule “restrict-template-expressions“